onnxruntime/contrib_ops/cuda/inverse.cc

```diff
@@ -35,22 +35,25 @@ ONNX_OPERATOR_KERNEL_EX(
 namespace inverse_internal {
 
 template <typename T>
-Status ComputeMatrixOffsets(T* workspace_data, size_t num_batches, size_t rows, IAllocatorUniquePtr<T*>& matrix_ptrs) {
+Status ComputeMatrixOffsets(cudaStream_t stream, T* workspace_data, size_t num_batches, size_t rows, IAllocatorUniquePtr<T*>& matrix_ptrs) {
   std::vector<T*> cuda_ptrs;
   const size_t matrix_size = rows * rows;
   for (size_t i = 0; i < num_batches; ++i) {
     cuda_ptrs.push_back(workspace_data);
     workspace_data += matrix_size;
   }
-  CUDA_RETURN_IF_ERROR(cudaMemcpy(matrix_ptrs.get(), cuda_ptrs.data(), sizeof(T*) * num_batches,
-                                  cudaMemcpyHostToDevice));
+
+  CUDA_RETURN_IF_ERROR(cudaStreamSynchronize(stream));
+  CUDA_RETURN_IF_ERROR(cudaMemcpyAsync(matrix_ptrs.get(), cuda_ptrs.data(), sizeof(T*) * num_batches,
```

Comment:
>cudaMemcpyAsync [](start = 23, length = 15)

This is no use. cuda_ptrs is not in pinned memory, so it will still be blocking. Need to use CudaAsyncBuffer if async copy is preferred.